apps/meteor/ee/app/livechat-enterprise/server/api/contacts.ts

```diff
@@ -0,0 +1,84 @@
+import Ajv from 'ajv';
+
+import { API } from '../../../../../app/api/server';
+import { changeContactBlockStatus, closeBlockedRoom, ensureSingleContactLicense } from './lib/contacts';
+
+const ajv = new Ajv({
+	coerceTypes: true,
+});
+
+type blockContactProps = {
+	visitorId: string;
+};
+
+const blockContactSchema = {
+	type: 'object',
+	properties: {
+		visitorId: {
+			type: 'string',
+		},
+	},
+	required: ['visitorId'],
+	additionalProperties: false,
+};
+
+const isBlockContactProps = ajv.compile<blockContactProps>(blockContactSchema);
+
+declare module '@rocket.chat/rest-typings' {
+	// eslint-disable-next-line @typescript-eslint/naming-convention
+	interface Endpoints {
+		'/v1/omnichannel/contacts.block': {
+			POST: (params: blockContactProps) => void;
+		};
+		'/v1/omnichannel/contacts.unblock': {
+			POST: (params: blockContactProps) => void;
+		};
+	}
+}
+
+API.v1.addRoute(
+	'omnichannel/contacts.block',
+	{
+		authRequired: true,
+		permissionsRequired: ['block-livechat-contact'],
+		validateParams: isBlockContactProps,
+	},
+	{
+		async post() {
+			ensureSingleContactLicense();
+			const { visitorId } = this.bodyParams;
+			const { user } = this;
+
+			await changeContactBlockStatus({
```

Comment:
I recommend there should be at least an info log that someone blocked a contact. Something like: `$userId blocked contact $id` for "audit" purposes.

This way we won't have the situation of "a contact was blocked out of nowhere", since we don't have a history of who blocked a contact.